arch/risc-v/src/common/riscv_percpu.c

```diff
@@ -61,11 +61,11 @@ static struct riscv_percpu_s g_scratch[HART_CNT];
 
 void riscv_percpu_init(void)
 {
-  int i;
+  uintptr_t i;
```

Comment:
I think using int as iterator is quite common in the code, but using stdint types is great IMO

Reply:
I change i to uintptr_t because hartid is uintptr_t too.